File: Tree.js
class Node {
    constructor(data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

class Tree {
    constructor() {
        this.root = null;
    }

    buildTree(x) {
        const newNode = new Node(x);

        
        if (this.root === null) {
            this.root = newNode;
            return;
        }

       
        const q = [];
        q.push(this.root);

        while (q.length > 0) {
            let temp = q.shift();

            if (temp.left === null) {
                temp.left = newNode;
                return;
            } else {
                q.push(temp.left);
            }

            if (temp.right === null) {
                temp.right = newNode;
                return;
            } else {
                q.push(temp.right);
            }
        }
    }

    levelOrder() {
        if (!this.root) return;

        const q = [];
        q.push(this.root);

        while (q.length > 0) {
            const node = q.shift();
            process.stdout.write(node.data + " ");

            if (node.left) q.push(node.left);
            if (node.right) q.push(node.right);
        }
    }
}


const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const tree = new Tree();
let inputCount = 0;
let expectedInputs = 0;

console.log("Enter number of nodes: ");

rl.on("line", (line) => {
    if (expectedInputs === 0) {
        expectedInputs = parseInt(line.trim());
        console.log("Enter node values:");
    } else {
        tree.buildTree(parseInt(line.trim()));
        inputCount++;

        if (inputCount === expectedInputs) {
            console.log("Level-order traversal:");
            tree.levelOrder();
            rl.close();
        }
    }
});

File: tree.py
from collections import *

class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

class Tree:
    def __init__(self):
        self.root = None

    def build_tree(self, x):
        new_node = Node(x)

       
        if self.root is None:
            self.root = new_node
            return

      
        q = deque()
        q.append(self.root)

        while q:
            temp = q.popleft()

            if temp.left is None:
                temp.left = new_node
                return
            else:
                q.append(temp.left)

            if temp.right is None:
                temp.right = new_node
                return
            else:
                q.append(temp.right)

    def level_order(self):
        if self.root is None:
            return

        q = deque()
        q.append(self.root)

        while q:
            node = q.popleft()
            print(node.data, end=' ')
            if node.left:
                q.append(node.left)
            if node.right:
                q.append(node.right)


# Main like section
if __name__ == "__main__":
    t = Tree()
    n = int(input("Enter number of nodes: "))
    print("Enter node values:")
    for _ in range(n):
        val = int(input())
        t.build_tree(val)

    print("Level-order traversal:")
    t.level_order()
